src/Jupyter/CustomShell/CustomShellRouter.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Jupyter.Core.Protocol;
+
+namespace Microsoft.Quantum.IQSharp.Jupyter
+{
+
+    public class CustomShellRouter : ICustomShellRouter
```

Comment:
I think it would make sense to push this to the `Core`, no?

Reply:
i.e. push `ICustomShellRouter` and `ICustomShellHandler` to Jupyter.Core.

Reply:
Good idea, I'll go on and do that in microsoft/jupyter-core#43, then.

Reply:
Resolving now that microsoft/jupyter-core#43 is completed. Thanks for the suggestion!